pkg/controller/healthcheck/healtcheck_actuator.go

```diff
@@ -93,28 +96,73 @@ type checkResultForConditionType struct {
 func (a *Actuator) ExecuteHealthCheckFunctions(ctx context.Context, request types.NamespacedName) (*[]Result, error) {
 	_, shootClient, err := util.NewClientForShoot(ctx, a.seedClient, request.Namespace, client.Options{})
 	if err != nil {
-		msg := fmt.Sprintf("failed to create shoot client in namespace '%s'", request.Namespace)
-		a.logger.Error(err, msg)
-		return nil, fmt.Errorf(msg)
+		msg := fmt.Errorf("failed to create shoot client in namespace '%s'", request.Namespace)
+		a.logger.Error(err, msg.Error())
+		return nil, msg
 	}
-	channel := make(chan channelResult)
-	var wg sync.WaitGroup
-	wg.Add(len(a.healthCheckMappings))
-	for healthCheck, healthConditionType := range a.healthCheckMappings {
+
+	var (
+		channel = make(chan channelResult)
+		wg      sync.WaitGroup
+	)
+
+	wg.Add(len(a.healthChecks))
+	for _, hc := range a.healthChecks {
 		// clone to avoid problems during parallel execution
-		check := healthCheck.DeepCopy()
+		check := hc.HealthCheck.DeepCopy()
 		check.InjectSeedClient(a.seedClient)
 		check.InjectShootClient(shootClient)
 		check.SetLoggerSuffix(a.provider, a.extensionKind)
-		go func(ctx context.Context, request types.NamespacedName, check HealthCheck, healthConditionType string) {
+
+		go func(ctx context.Context, request types.NamespacedName, check HealthCheck, preCheckFunc PreCheckFunc, healthConditionType string) {
 			defer wg.Done()
+
+			obj := a.getExtensionObjFunc()
+			if err := a.seedClient.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, obj); err != nil {
+				channel <- channelResult{
+					healthCheckResult: &SingleCheckResult{
+						IsHealthy: false,
+						Detail:    err.Error(),
+						Reason:    "ReadExtensionObjectFailed",
+					},
+					error:               err,
+					healthConditionType: healthConditionType,
+				}
+				return
+			}
+
+			cluster, err := extensionscontroller.GetCluster(ctx, a.seedClient, request.Namespace)
+			if err != nil {
+				channel <- channelResult{
+					healthCheckResult: &SingleCheckResult{
+						IsHealthy: false,
+						Detail:    err.Error(),
+						Reason:    "ReadClusterObjectFailed",
+					},
+					error:               err,
+					healthConditionType: healthConditionType,
+				}
+				return
+			}
+
+			if preCheckFunc != nil && !preCheckFunc(obj, cluster) {
+				channel <- channelResult{
+					healthCheckResult: &SingleCheckResult{
+						IsHealthy: true,
```

Comment:
I get the motivation setting this to `true` but in the future we should think about being more transparent, i.e. introducing a `ConditionStatus` that is acceptable/expected like `skipped`. For now we can at least log that the health check is skipped, so an operator gets this type of transparency. WDYT?

Reply:
I can add the log, yes. The new condition status is a larger topic, indeed. In the g/g core care controller we also set it to `True` when the shoot is hibernated and no checks are performed.